### PR TITLE
Support for local cdrom storage in vagrant-libvirt

### PR DESCRIPTION
**Pull request by d0c-s4vage**
Support for local cdrom storage using:

    config.vm.provider :libvirt do |libvirt|
        libvirt.storage :file, :device => :cdrom, :path => "/some/path/to/cdrom.iso"
    end

### CODE REVIEW

File: lib/vagrant-libvirt/action/create_domain.rb [hunk 1]
```diff
@@ -52,6 +52,7 @@ def call(env)
           # Storage
           @storage_pool_name = config.storage_pool_name
           @disks = config.disks
+		  @cdrom = config.cdrom
```
Comment (pronix) [inline lib/vagrant-libvirt/action/create_domain.rb]: allign it please

File: lib/vagrant-libvirt/config.rb [hunk 2]
```diff
@@ -129,6 +131,40 @@ def _get_device(disks)
...
+      def _handle_file_storage(options={})
+        if options[:device] == :cdrom
+          _handle_cdrom_storage(options)
+        else
+          _handle_disk_storage(options)
+        end
+      end
+
+      def _handle_cdrom_storage(options={})
+        # <disk type="file" device="cdrom">
```
Comment (pronix) [inline lib/vagrant-libvirt/config.rb]: remove it please if not need.

File: lib/vagrant-libvirt/config.rb [hunk 3]
```diff
@@ -129,6 +131,40 @@ def _get_device(disks)
...
+        #   <readonly/>
+        # </disk>
+
+        options = {
+          :dev => "hdc",
+          :bus => "ide",
+          :path => nil,
+        }.merge(options)
+
+        @cdrom = {
```
Comment (pronix) [inline lib/vagrant-libvirt/config.rb]: why not multiple cdroms ? (as disks) or it is not possible ?

Comment (d0c-s4vage) [inline lib/vagrant-libvirt/config.rb]: eh, I hadn't tested it, and a single one fit my use case. Multiple cdroms should work. I'll test it out

**Comment by d0c-s4vage** [comment]
In my testing up to four cdroms could be created (hda,hdb,hdc,hdd):

        user [ blah3 ]: vagrant up
        Bringing machine 'default' up with 'libvirt' provider...
        ==> default: Creating image (snapshot of base box volume).
        ==> default: Creating domain with the following settings...
        ==> default:  -- Name:              blah3_default
        ==> default:  -- Domain type:       kvm
        ==> default:  -- Cpus:              1
        ==> default:  -- Memory:            512M
        ==> default:  -- Base box:          testing
        ==> default:  -- Storage pool:      default
        ==> default:  -- Image:             /var/lib/libvirt/images/blah3_default.img
        ==> default:  -- Volume Cache:      default
        ==> default:  -- Kernel:
        ==> default:  -- Initrd:
        ==> default:  -- Graphics Type:     vnc
        ==> default:  -- Graphics Port:     5900
        ==> default:  -- Graphics IP:       127.0.0.1
        ==> default:  -- Graphics Password: Not defined
        ==> default:  -- Video Type:        cirrus
        ==> default:  -- Video VRAM:        9216
        ==> default:  -- Keymap:            en-us
        ==> default:  -- CDROMS:            hda, hdb, hdc, hdd
        ==> default:  -- CDROM(hda):        /tmp/virtio-drivers.iso
        ==> default:  -- CDROM(hdb):        /tmp/virtio-drivers.iso
        ==> default:  -- CDROM(hdc):        /tmp/virtio-drivers.iso
        ==> default:  -- CDROM(hdd):        /tmp/virtio-drivers.iso
        ==> default:  -- Command line :
        ==> default: Starting domain.
        ==> default: Waiting for domain to get an IP address...

**Comment by pronix** [comment]
if it possible, please add information to README also

**Comment by pronix** [comment]
if it possible, please add information to README also

**Comment by pronix** [comment]
if it possible, please add information to README also

**Pull request merged**